### DISCUSSION TITLE
Deploying a vanilla JS SPA

### DISCUSSION THREAD

**Discussion by chrishrtmn** [vercel/community#1042] [Help] [answered]
Are there any examples or tutorials of a SEO-friendly SPA with routes using vanilla JS (without frameworks) on Vercel?

**Comment by eduhsoto** [comment]
Hey, Do you found help?

**Comment by amyegan** [comment, marked as answer]
With the new [Build Output API](https://vercel.com/blog/build-output-api), projects using any framework (or no framework!) can be built for Vercel. Building and SEO-friendly SPA with routes using vanilla JS is more of a code question, so that would be up to you to architect. But there are quite a few [examples of Build Output API solutions](https://github.com/vercel/examples/tree/main/build-output-api) in the examples repo.

**Comment by walidChaybi** [comment]
function init() {
  switch (global.currentPgae) {
    case "/":
    case "/index.html":
      displayPopularMovies();
      displaySlider();
      break;
    case "/shows.html":
      displayPopularShows();
      break;
    case "/movie-details.html":
      displayMovieDetails();
      break;
    case "/tv-details.html":
      displayShowDetails();
      break;
    case "/search.html":
      search();
      break;
  }
  highlightActiveLink();
}

you can use this template to Routing, you just need to grap the current page info from the window object
